File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.java
class Solution {
    PriorityQueue<Integer> minheap = new PriorityQueue<Integer>(3);
    
    public int maxProduct(int[] nums) {
        for (int i = 0; i < nums.length; i++ ) {
            if (minheap.size() < 3) {
                minheap.add(nums[i]);
            } else {
                minheap.poll();
                minheap.add(nums[i]);
            }
        }
        if (minheap.size() == 2) {
            return (minheap.poll()-1)*(minheap.poll()-1);
        }
        
        minheap.poll();
        return (minheap.poll()-1)*(minheap.poll()-1);
    }
    
}

File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.py
class Solution(object):
    def maxProduct(self, nums):
        max = 0
        for i in range(len(nums)-1):
            for j in range(i+1, len(nums)):
                if (nums[i]-1)* (nums[j]-1) > max:
                    max = (nums[i]-1)* (nums[j]-1)
        return max
                    